### PR TITLE
Add resize feature for pattern previews

### PR DESCRIPTION
**Pull request by olaseni**
Related to #5602

This feature wraps pattern previews  with a `ResizableBox` to enable resizing.

Load `/patterns/about` and confirm that you can see the resizable handle at the right side of the pattern preview.

Screenshot 2024-03-17 at 06 33 18

### CODE REVIEW

File: client/my-sites/patterns/components/pattern-preview/index.tsx [hunk 1]
```diff
@@ -31,23 +32,39 @@ export function PatternPreview( {
...
+				right: true,
+				bottom: false,
+				left: false,
+				topRight: false,
+				bottomRight: false,
+				bottomLeft: false,
+				topLeft: false,
+			} }
+			handleWrapperClass={'pattern-preview__resizer'}
+			minWidth={500}
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/index.tsx]:
```suggestion
			minWidth={375}
```
This is closer to a modern mobile viewport (iPhone 15 has a 393px wide viewport).

File: client/my-sites/patterns/components/pattern-preview/index.tsx [hunk 2]
```diff
@@ -31,23 +32,39 @@ export function PatternPreview( {
 	}
 
 	return (
-		<div
-			className={ classNames( 'pattern-preview', {
-				'pattern-preview_category-gallery': isCategoryPreview,
-				'is-loading': ! renderedPattern,
-			} ) }
+		<ResizableBox
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/index.tsx:117]: The `ResizableBox` wrapper seems to mess up the pattern previews in the category links on the `/patterns` page. We should do something about that.

image

File: client/my-sites/patterns/components/pattern-preview/index.tsx [hunk 3]
```diff
@@ -31,23 +32,39 @@ export function PatternPreview( {
...
+				top: false,
+				right: true,
+				bottom: false,
+				left: false,
+				topRight: false,
+				bottomRight: false,
+				bottomLeft: false,
+				topLeft: false,
+			} }
+			handleWrapperClass={'pattern-preview__resizer'}
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/index.tsx]:
```suggestion
			handleWrapperClass="pattern-preview__resizer"
```
Nit

File: client/my-sites/patterns/components/pattern-preview/index.tsx (hunk 2 above)
Comment (fredrikekelund) [reply client/my-sites/patterns/components/pattern-preview/index.tsx:117]: We should also disable the resizing entirely in grid view. 
grid view

**Review by fredrikekelund — CHANGES_REQUESTED** [review]
Thanks for taking care of this, @olaseni 👍 I left a few comments. Most importantly, we should fix the pattern previews in the category links and disable the resizing in grid view mode.

I also think we should consider center aligning pattern previews instead of left aligning them (the way it currently works in this PR). @matt-west could you confirm if this was your intention with the design?

**Comment by matt-west** [comment]
> I also think we should consider center aligning pattern previews instead of left aligning them (the way it currently works in this PR). @matt-west could you confirm if this was your intention with the design?

Do you mean in the grid view?

**Comment by fredrikekelund** [comment]
@matt-west, I mean in the list view. 

Like this:

| Current behavior | My suggestion |
| - | - |
| current | suggestion |

**Comment by matt-west** [comment]
Ah, thank you for clarifying.

Let's keep it left-aligned. Given that the resize handle is on the right, it feels like the pattern should stay pinned to the left side while resizing.

File: client/my-sites/patterns/components/pattern-preview/index.tsx (hunk 2 above)
Comment (olaseni) [reply client/my-sites/patterns/components/pattern-preview/index.tsx:117]: > We should also disable the resizing entirely in grid view.

Done.

> The ResizableBox wrapper seems to mess up the pattern previews in the category links on the /patterns page. We should do something about that.

It looks like it's not needed in that case either. So I have added an `isResizable` prop to actively indicate intent to resize

File: client/my-sites/patterns/components/pattern-preview/style.scss [hunk 4]
```diff
@@ -33,3 +33,21 @@
...
 }
+
+.pattern-preview__resizer .components-resizable-box__handle-right {
+	&::after {
+		background-color: #a7aaad;
+		border-radius: 3px;
+		box-shadow: none;
+		height: 50px;
+		margin-right: -10px;
+		right: calc(50% - 5px);
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/style.scss:125]:
```suggestion
		right: calc(50% - 5px);
		top: calc(50% - 8px - 3rem);
```
The value I've used here isn't ideal, but the idea is to center the handle to the actual preview viewport, as opposed to centering it to the entire `.pattern-preview` (which also includes the title).

File: client/my-sites/patterns/components/pattern-preview/index.tsx [hunk 5]
```diff
@@ -12,42 +13,75 @@ export const ASPECT_RATIO = 7 / 4;
...
 	viewportWidth,
-}: PatternPreviewProps ) {
+}: PatternPreviewProps) {
 	const { renderedPatterns } = usePatternsRendererContext();
-	const patternId = encodePatternId( pattern?.ID ?? 0 );
-	const renderedPattern = renderedPatterns[ patternId ];
-	const [ resizeObserver, nodeSize ] = useResizeObserver();
+	const patternId = encodePatternId(pattern?.ID ?? 0);
+	const renderedPattern = renderedPatterns[patternId];
+	const [resizeObserver, nodeSize] = useResizeObserver();
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-preview/index.tsx]: It looks like your ESLint/Prettier is acting up 🤔 Those extra spaces should be preserved

File: client/my-sites/patterns/components/pattern-gallery/client.tsx [hunk 6]
```diff
@@ -28,6 +28,7 @@ export const PatternGalleryClient: PatternGalleryFC = ( { isGridView, patterns =
 				<div className={ classNames( 'pattern-gallery', { 'pattern-gallery--grid': isGridView } ) }>
 					{ patterns.map( ( pattern ) => (
 						<PatternPreview
+							isResizable={!isGridView}
```
Comment (fredrikekelund) [inline client/my-sites/patterns/components/pattern-gallery/client.tsx]: Also ESLint/Prettier

**Review by fredrikekelund — APPROVED** [review]
Looks great overall! 👍 Left some comments that we should address before landing, but it's all small stuff, so I'll approve straight away

File: client/my-sites/patterns/components/pattern-preview/index.tsx (hunk 5 above)
Comment (olaseni) [reply client/my-sites/patterns/components/pattern-preview/index.tsx]: Yes, it looks like I got the default formatter. I'll fix that.

**Pull request merged**